tests/testthat/test-layers.R

```diff
@@ -31,16 +33,214 @@ test_that("time distributed layer works", {
 
 })
 
-test_that("gated residual network", {
+test_that("gated residual network, w and wo output_size, w or wo dropout_rate, w or wo return_gate", {
 
-  x <- torch::torch_randn(32, 10)
-  grn <- gated_residual_network(10, 5)
+  hidden_layer_size <- 16
+  num_static <- 3
+  # 2-dim x
+  x <- torch::torch_randn(32, 10, device=device)
+  grn <- gated_residual_network(10, 5)$to(device=device)
+
+  expect_equal(grn(x)$shape, c(32,5))
+
+  x <- torch::torch_randn(32, 16, device=device)
+  static_context_grn <- gated_residual_network(hidden_layer_size, hidden_layer_size)$to(device=device)
+
+  expect_equal(static_context_grn(x)$shape, x$shape)
+
+  # 3-dim x
+  x <- torch::torch_randn(12, 1, hidden_layer_size*num_static, device=device) # [12, 1, 48]
+  grn <- gated_residual_network(hidden_layer_size*num_static,
+                                hidden_layer_size,
+                                num_static)$to(device=device) # [48, 16, 3]
+
+  expect_equal(grn(x)$shape, c(12, 1, num_static)) # [12, 1, 3]
+
+  grn <- gated_residual_network(48, 16, 3, dropout_rate=0.2)$to(device=device)
+
+  expect_equal(grn(x)$shape, c(12, 1, 3))
+
+  grn <- gated_residual_network(48, 16, 3, return_gate=TRUE)$to(device=device)
+
+  expect_equal(grn(x)[[1]]$shape, c(12, 1, 3))
+  expect_equal(grn(x)[[2]]$shape, c(12, 1, 3))
+
+
+})
+
+test_that("gated residual network works in all initial conditions w additional_context", {
+
+  x <- torch::torch_randn(4, 19, 16*3, device=device)
+  context <- torch::torch_ones(4, 1, 19)$to(device=device)
```

Comment:
@cregouby it feels to me that context should be size (4,1,16), ie. it's dimension should be the same as `hidden_layer_size` in the gated_residual_network initialization. Are we sure this is the right starting dimention?